drivers/mfd/rp1.c

```diff
@@ -74,6 +74,20 @@ static int rp1_probe(struct udevice *dev)
 	/* Turn on bus-mastering */
 	dm_pci_clrset_config16(dev, PCI_COMMAND, 0, PCI_COMMAND_MASTER | PCI_COMMAND_MEMORY);
 
+	/*
+	 * HACK: Set bar address in the correct order. RP1 driver in Linux Kernel rely on
+	 * the pci BAR configuration which initializes BARs based on BAR size.
+	 * This results the Linux set BARs in the same order on each boot.
+	 * U-boot does initialization without any sorting so the configuration
+	 * may be different from the Linux kernel.
+	 * Here BAR address configuration is set to match the Linux Kernel order
+	 * to avoid fails on address translation.
+	 * This should be changed after upstreaming RP1 driver to the Linux kernel
+	 * mainline.
+	 */
+	dm_pci_write_config32(dev, PCI_BASE_ADDRESS_1, 0);
```

Comment:
I believe, in my setup all was working without this code. Could you please check?

Reply:
unfortunatelly no. It doesn't work without this fix

Reply:
Okay then. We now have procedure for HACKs: https://xen-troops.github.io/process/process under "Temporary changes" section. Please follow it.

Reply:
updated according to the process description